src/jdk.compiler/share/classes/com/sun/tools/javac/model/JavacElements.java

```diff
@@ -199,42 +200,44 @@ private ClassSymbol doGetTypeElement(ModuleElement module, CharSequence name) {
         return (S) resultCache.computeIfAbsent(Pair.of(methodName, nameStr), p -> {
             Set<S> found = new LinkedHashSet<>();
 
-            for (ModuleSymbol msym : modules.allModules()) {
-                S sym = nameToSymbol(msym, nameStr, clazz);
+            for (Set<ModuleSymbol> allModules : Arrays.asList(modules.getRootModules(), modules.allModules())) {
```

Comment:
does it make sense to remove the root modules from the all modules set in order to avoid searching twice in the root modules if the symbol is not found there?